internal/global/common/weakness_break.go

```diff
@@ -0,0 +1,194 @@
+
+	if /* enemyInfo.IsElite */ true {
+		windShearStacksCount = 3
+	} else {
+		windShearStacksCount = 1
+	}
+
+	engine.AddModifier(enemyID, info.Modifier{
+		Name:     BreakWindShear,
+		Source:   characterID,
+		Duration: 2,
+		Count:    windShearStacksCount,
+		Chance:   1.5,
+	})
+}
+
+func applyWeaknessBreakQuantum(engine engine.Engine, characterID, enemyID key.TargetID, characterInfo info.Character, enemyInfo info.Enemy, maxStanceMultiplier float64) {
+	dealWeaknessBreakDamage(engine, WeaknessBreakQuantum, characterID, enemyID, characterInfo.Element, maxStanceMultiplier, 0.5)
+	modifyGaugeAfterWeaknessBreak(engine, WeaknessBreakQuantum, characterID, enemyID)
+
+	engine.AddModifier(enemyID, info.Modifier{
+		Name:   BreakEntanglement,
+		Source: characterID,
+		State: &BreakEntanglementState{
+			HitsTakenCount:            0,
+			TargetMaxStanceMultiplier: maxStanceMultiplier,
+		},
+		Duration: 1,
+		Chance:   1.5,
+	})
+}
+
+func applyWeaknessBreakImaginary(engine engine.Engine, characterID, enemyID key.TargetID, characterInfo info.Character, enemyInfo info.Enemy, maxStanceMultiplier float64) {
+	dealWeaknessBreakDamage(engine, WeaknessBreakImaginary, characterID, enemyID, characterInfo.Element, maxStanceMultiplier, 0.5)
+	modifyGaugeAfterWeaknessBreak(engine, WeaknessBreakImaginary, characterID, enemyID)
+
+	engine.AddModifier(enemyID, info.Modifier{
+		Name:   BreakImprisonment,
+		Source: characterID,
+		State: &BreakImprisonState{
+			DelayRatio: 0.2 * (1 + engine.Stats(characterID).BreakEffect()),
+		},
+		Duration: 1,
+		Chance:   1.5,
+	})
+}
+
+func ApplyWeaknessBreakEffects(engine engine.Engine, characterID, enemyID key.TargetID) {
+	characterInfo, err := engine.CharacterInfo(characterID)
+
+	if err != nil {
+		panic("incorrect target id for character")
+	}
+
+	enemyInfo, err := engine.EnemyInfo(enemyID)
+
+	if err != nil {
+		panic("incorrect target id for enemy")
+	}
+
+	damageType := characterInfo.Element
+	maxStanceMultiplier := 0.5 + enemyInfo.MaxStance/120
+
+	switch damageType {
+	case model.DamageType_PHYSICAL:
+		applyWeaknessBreakPhysical(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_FIRE:
+		applyWeaknessBreakFire(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_ICE:
+		applyWeaknessBreakIce(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_THUNDER:
+		applyWeaknessBreakThunder(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_WIND:
+		applyWeaknessBreakWind(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_QUANTUM:
+		applyWeaknessBreakQuantum(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	case model.DamageType_IMAGINARY:
+		applyWeaknessBreakImaginary(engine, characterID, enemyID, characterInfo, enemyInfo, maxStanceMultiplier)
+	}
```

Comment:
given the number of params and how they are all identical, might be cleaner to construct and pass in a struct instead:

```go
type breakInfo struct {
  engine              engine.Engine
  charID              key.TargetID
  charInfo            *info.Character
  enemyID             key.TargetID
  enemyInfo           *info.Character
  maxStanceMultiplier float64
}

func ApplyWeaknessBreakEffects(engine engine.Engine, characterID, enemyID key.TargetID) {
	characterInfo, err := engine.CharacterInfo(characterID)
	if err != nil {
		panic("incorrect target id for character")
	}

	enemyInfo, err := engine.EnemyInfo(enemyID)
	if err != nil {
		panic("incorrect target id for enemy")
	}

  break := &breakInfo{
    engine:              engine,
    charID:              characterID,
    charInfo:            &characterInfo,
    enemyID:             enemyID,
    enemyInfo:           &enemyInfo,
    maxStanceMultiplier: 0.5 + enemyInfo.MaxStance/120,
  }

  switch characterInfo.Element {
  case model.DamageType_PHYSICAL:
		applyWeaknessBreakPhysical(break)
	case model.DamageType_FIRE:
		applyWeaknessBreakFire(break)
	case model.DamageType_ICE:
		applyWeaknessBreakIce(break)
	case model.DamageType_THUNDER:
		applyWeaknessBreakThunder(break)
	case model.DamageType_WIND:
		applyWeaknessBreakWind(break)
	case model.DamageType_QUANTUM:
		applyWeaknessBreakQuantum(break)
	case model.DamageType_IMAGINARY:
		applyWeaknessBreakImaginary(break)
	}
}
```

Additionally `dealWeaknessBreakDamage` can then be simplified to the signature of
```go
func dealWeaknessBreakDamage(data *breakInfo, damageMultiplier float64)
```

could also make all these functions a function of `breakInfo` to change calls to `break.applyWeaknessBreakWind()` and `break.dealWeaknessBreakDamage(dmgMultiplier float64)`